### PR TITLE
Propagating Custom certs referenced in pachd's SSL_CERT_DIR to side-cars

### PR DESCRIPTION
**Pull request by acohen4**

### CODE REVIEW

File: etc/helm/pachyderm/templates/pachd/deployment.yaml [hunk 1]
```diff
@@ -150,6 +150,10 @@ spec:
         - name: SSL_CERT_DIR
           value:  /pachd-tls-cert
         {{- end }}
+        {{- if .Values.pachd.tls.enabled }}
```
Comment (tybritten) [inline etc/helm/pachyderm/templates/pachd/deployment.yaml:153]: shouldn't this be if global.customcacerts?

**Review by tybritten — COMMENTED** [review]
just one question. wasn't sure if you want to set that value unless customcacert is set

File: etc/helm/pachyderm/templates/pachd/deployment.yaml (hunk 1 above)
Comment (acohen4) [reply etc/helm/pachyderm/templates/pachd/deployment.yaml:153]: I think either this or `{{- if and .Values.pachd.tls.enabled .Values.global.customCaCerts }}` as we have elsewhere should work. Right now the only use-case to pass the secret name to pachd is for the `customCaCerts` case, but this value is more general. I can change it to be consistent

Comment (tybritten) [reply etc/helm/pachyderm/templates/pachd/deployment.yaml]: Lgtm. Testing locally now. I'll package up my setup so we can make a permanent test

**Review by tybritten — APPROVED** [review]
Tested this locally and works as intended.

File: src/internal/serviceenv/config.go [hunk 2]
```diff
@@ -98,6 +98,7 @@ type PachdSpecificConfiguration struct {
 	// TODO: Merge this with the worker specific pod name (PPS_POD_NAME) into a global configuration pod name.
 	PachdPodName                 string `env:"PACHD_POD_NAME,required"`
 	EnableWorkerSecurityContexts bool   `env:"ENABLE_WORKER_SECURITY_CONTEXTS,default=true"`
+	TLSCertsSecretName           string `env:"TLS_CERT_SECRET_NAME,default="`
```
Comment (albscui) [inline src/internal/serviceenv/config.go:101]:
```suggestion
	TLSCertSecretName           string `env:"TLS_CERT_SECRET_NAME,default="`
```

**Review by albscui — APPROVED** [review]
LGTM! Just a nit on the `TLSCertsSecretName`.

File: src/server/pps/server/worker_rc.go [hunk 3]
```diff
@@ -88,6 +88,22 @@ func getPachctlSecretVolumeAndMount(secret string) (v1.Volume, v1.VolumeMount) {
 		}
 }
 
+// getTLSCertSecretVolumeAndMount returns a Volume and VolumeMount object
+// configured for the pach-tls secret to be stored in pipeline side-cars.
+func getTLSCertSecretVolumeAndMount(secret, mountPath string) (v1.Volume, v1.VolumeMount) {
+	return v1.Volume{
+			Name: secret,
```
Comment (albscui) [inline src/server/pps/server/worker_rc.go:95]: Is name of the volume something we control? Maybe call it something like `<secret>-volume` to distinguish it from the actual secret name?

Comment (acohen4) [reply src/server/pps/server/worker_rc.go:95]: it is something that we control. We could do that, but to be consistent with other volumes, mounts we're creating here, I think just using the secret name makes sense

**Pull request merged**